### PR TITLE
add a builder class to create the exporter

### PR DESCRIPTION
**Pull request by jkwatson**

### CODE REVIEW

File: exporters/newrelic/src/main/java/io/opentelemetry/exporters/newrelic/NewRelicSpanExporter.java [hunk 1]
```diff
@@ -70,4 +76,91 @@ public ResultCode export(List<Span> openTracingSpans) {
...
+      return this;
+    }
+
+    /**
+     * Set your New Relic Insert Key.
+     *
+     * @param apiKey your New Relic Insert Key.
+     * @return this builder's instance
+     */
+    public Builder newRelicApiKey(String apiKey) {
```
Comment (breedx-nr) [inline exporters/newrelic/src/main/java/io/opentelemetry/exporters/newrelic/NewRelicSpanExporter.java]: can we shorten this to just `insertKey` or `apiKey`?  The `newRelic` bit seems redundant.

File: exporters/newrelic/src/main/java/io/opentelemetry/exporters/newrelic/NewRelicSpanExporter.java [hunk 2]
```diff
@@ -70,4 +76,91 @@ public ResultCode export(List<Span> openTracingSpans) {
...
+      if (spanBatchSender == null) {
+        SpanBatchSenderBuilder builder = SimpleSpanBatchSender.builder(apiKey);
+        if (enableAuditLogging) {
+          builder.enableAuditLogging();
+        }
+        spanBatchSender = builder.build();
+      }
+      return new NewRelicSpanExporter(new SpanBatchAdapter(commonAttributes), spanBatchSender);
+    }
+  }
```
Comment (breedx-nr) [inline exporters/newrelic/src/main/java/io/opentelemetry/exporters/newrelic/NewRelicSpanExporter.java:165]: I'd love to have a sweet `uriOverride()` here as well for [staging] testing.

**Review by breedx-nr — CHANGES_REQUESTED** [review]
A few small asks.

File: exporters/newrelic/src/main/java/io/opentelemetry/exporters/newrelic/NewRelicSpanExporter.java (hunk 2 above)
Comment (jkwatson) [reply exporters/newrelic/src/main/java/io/opentelemetry/exporters/newrelic/NewRelicSpanExporter.java:165]: I was trying to avoid just forwarding our entire builder over to this one. :) You can always just provide your own `SpanBatchSender` with this builder, too.

File: exporters/newrelic/src/main/java/io/opentelemetry/exporters/newrelic/NewRelicSpanExporter.java (hunk 1 above)
Comment (jkwatson) [reply exporters/newrelic/src/main/java/io/opentelemetry/exporters/newrelic/NewRelicSpanExporter.java]: can do

File: exporters/newrelic/src/main/java/io/opentelemetry/exporters/newrelic/NewRelicSpanExporter.java (hunk 2 above)
Comment (breedx-nr) [reply exporters/newrelic/src/main/java/io/opentelemetry/exporters/newrelic/NewRelicSpanExporter.java:165]: Fair nuff.

File: exporters/newrelic/src/main/java/io/opentelemetry/exporters/newrelic/NewRelicSpanExporter.java (hunk 1 above)
Comment (jkwatson) [reply exporters/newrelic/src/main/java/io/opentelemetry/exporters/newrelic/NewRelicSpanExporter.java]: done

**Review by breedx-nr — APPROVED** [review]
